lib/s3routes/routes.js

```diff
@@ -52,8 +54,12 @@ function checkBucketAndKey(bucketName, objectKey, method, reqQuery,
             blacklistedPrefixes.object);
         if (!result.isValid) {
             log.debug('invalid object key', { objectKey });
-            return errors.InvalidArgument.customizeDescription('Object key ' +
-            `must not start with "${result.invalidPrefix}".`);
+            if (result.invalidPrefix) {
+                return errors.InvalidArgument.customizeDescription('Object ' +
+                `key must not start with "${result.invalidPrefix}".`);
+            }
+            return errors.KeyTooLong.customizeDescription('Your key is ' +
```

Comment:
`Object key is too long. Maximum number of bytes allowed in keys is ${OBJECT_KEY_BYTE_LIMIT}.`